src/Generator.Bind/Writers/BindingWriter.cs

```diff
@@ -175,15 +175,13 @@ private Task WriteExtensionWrapperFiles([NotNull] string extensionName, [NotNull
             var translatedExtensionName = _identifierTranslator.Translate(extensionName);
 
             var nativeSignatures = _profile.NativeSignatures
-                .Where(ns => ns.Extension == extensionName)
-                .Where(ns => ns.Categories.First() == category)
+                .Where(ns => ns.Extension == extensionName && ns.Categories.First() == category)
                 .OrderBy(s => s.Name)
                 .ThenBy(s => s.Parameters.Count)
                 .ToList();
 
             var overloads = _profile.Overloads
-                .Where(o => o.Extension == extensionName)
```

Comment:
Why the combination here? It's more readable split up.

Reply:
Because you're enumerating a collection twice for no reason.

Would you write?
```csharp
var someList = blahblah;
for (int i = someList.Count - 1; i >= 0; i --)
{
    if (predicate(someList[i]))
    {
        someList.RemoveAt[i];
    }
}
for (int i = someList.Count - 1; i >= 0; i --)
{
    if (anotherPredicate(someList[i]))
    {
        someList.RemoveAt[i];
    }
}
```

Or would you write:

```csharp
var someList = blahblah;
for (int i = someList.Count - 1; i >= 0; i --)
{
    if (predicate(someList[i]) || anotherPredicate(someList[i]))
    {
        someList.RemoveAt[i];
    }
}

Reply:
Well LinQ does not iterate twice though with Where statements. as on each iterator the where statement yields if the predicate is true, but the next Where can skip it while it stll did not iterate through the whole Enumerable.
So it's not as bad as iterating twice, but performance wise it's ofc worse, because it allocates an additional Enumerator and delegate(not that bad for things like this) and has a double the function calls...
Readability wise - in my opinion it is debatable here as well which one is better.

Reply:
Yah, fair enough :)